src/components/tooltip/tooltip.pw.tsx

```diff
@@ -0,0 +1,375 @@
+import React from "react";
+import { test, expect } from "@playwright/experimental-ct-react17";
+import {
+  TooltipComponent,
+  UncontrolledTooltipComponent,
+  Default,
+  Controlled,
+  Positioning,
+  FlipBehaviourOverrides,
+  LargeTooltip,
+  Types,
+  ColorOverrides,
+} from "./components.test-pw";
+import Box from "../../../src/components/box";
+import { TooltipPositions } from "../../../src/components/tooltip/tooltip.config";
+import {
+  checkAccessibility,
+  assertCssValueIsApproximately,
+} from "../../../playwright/support/helper";
+import { SIZE, COLOR, CHARACTERS } from "../../../playwright/support/constants";
+import { TooltipProps } from "../../../src/components/tooltip/tooltip.component";
+import { getDataElementByValue } from "../../../playwright/components";
+
+const backgroundColors = [COLOR.ORANGE, COLOR.RED, COLOR.BLACK, COLOR.BROWN];
+const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+
+test.describe("Tooltip component", () => {
+  testData.forEach((message) => {
+    test(`check when message prop is set as ${message}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TooltipComponent message={message} />);
+
+      await expect(getDataElementByValue(page, "tooltip")).toHaveText(message);
+    });
+  });
+
+  testData.forEach((id) => {
+    test(`check when id prop is set as ${id}`, async ({ mount, page }) => {
+      await mount(<TooltipComponent id={id} />);
+
+      await expect(getDataElementByValue(page, "tooltip")).toHaveId(id);
+    });
+  });
+
+  [true, false].forEach((boolVal) => {
+    test(`check when tooltip visibility is set as ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TooltipComponent isVisible={boolVal} />);
+
+      if (boolVal === true) {
+        await expect(getDataElementByValue(page, "tooltip")).toBeVisible();
+      } else {
+        await expect(getDataElementByValue(page, "tooltip")).not.toBeVisible();
+      }
+    });
+  });
+
+  (["bottom", "left", "right", "top"] as [
+    "bottom",
+    "left",
+    "right",
+    "top"
+  ]).forEach((position) => {
+    test(`check when tooltip position is set as ${position}`, async ({
```

Comment:
same as all the "is set as" below for the test description, can be changed to "is set to"